shared/bls/common/constants.go

```diff
@@ -0,0 +1,27 @@
+package common
+
+import "bytes"
+
+// ZeroSecretKey represents a zero secret key.
+var ZeroSecretKey = [32]byte{}
+
+// InfinitePublicKey represents an infinite public key.
+var InfinitePublicKey = [48]byte{0xC0}
+
+// InfiniteSignature represents an infinite signature.
+var InfiniteSignature = [96]byte{0xC0}
+
+// SecretKeyIsZero checks the validity of a secret key.
+func SecretKeyIsZero(key []byte) bool {
+	return bytes.Equal(key, ZeroSecretKey[:])
+}
+
+// PublicKeyIsInfinite checks the validity of a public key.
+func PublicKeyIsInfinite(key []byte) bool {
+	return bytes.Equal(key, InfinitePublicKey[:])
+}
```

Comment:
Is there a faster way to do this based on each library? I think marshalling to bytes is pretty slow, but it looks like you can complete this check in herumi like `pubKey.(*PublicKey).p.IsZero()`. Is that the same?

If so, you can accept some pubkey interface and call IsInfinite where the underlying library has support to detect this without marshalling to bytes. 

Same feedback for each of these three methods.